jdbc/src/main/java/org/apache/zeppelin/jdbc/JDBCInterpreter.java

```diff
@@ -618,21 +620,54 @@ private InterpreterResult executeSql(String propertyKey, String sql,
       }
       getJDBCConfiguration(user).removeStatement(paragraphId);
     } catch (Exception e) {
-      logger.error("Cannot run " + sql, e);
-      ByteArrayOutputStream baos = new ByteArrayOutputStream();
-      PrintStream ps = new PrintStream(baos);
-      e.printStackTrace(ps);
-      String errorMsg = new String(baos.toByteArray(), StandardCharsets.UTF_8);
+      if (e.getCause() instanceof TTransportException &&
+          e.getCause().getMessage().equals("GSS initiate failed") &&
+          getJDBCConfiguration(user).isConnectionInDBDriverPool(propertyKey)) {
+        return reLoginFromKeytab(propertyKey, sql, interpreterContext, interpreterResult);
+      } else {
+        logger.error("Cannot run " + sql, e);
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        PrintStream ps = new PrintStream(baos);
+        e.printStackTrace(ps);
+        String errorMsg = new String(baos.toByteArray(), StandardCharsets.UTF_8);
+        try {
+          closeDBPool(user, propertyKey);
+        } catch (SQLException e1) {
+          logger.error("Cannot close DBPool for user, propertyKey: " + user + propertyKey, e1);
+        }
+        interpreterResult.add(errorMsg);
+        return new InterpreterResult(Code.ERROR, interpreterResult.message());
+      }
+    }
+    return interpreterResult;
+  }
 
+  private InterpreterResult reLoginFromKeytab(String propertyKey, String sql,
+     InterpreterContext interpreterContext, InterpreterResult interpreterResult) {
+    String user = interpreterContext.getAuthenticationInfo().getUser();
+    try {
+      closeDBPool(user, propertyKey);
+    } catch (SQLException e) {/*ignored*/}
```

Comment:
I thought since, this in under reLogin and we don't care about if we were able to successfully  close the connection or not, hence, ignored. But, yes, I'll add some exception here, user may want to know about this.

I'll add exception here.